### ISSUE TITLE
Rule Proposal: Catch no await inside a catch block

### ISSUE THREAD

**Issue by erlichmen** [typescript-eslint/typescript-eslint#2821]
consider the following code:

```typescript
async function foo(val): number {
  if (!val) {
     throw new Error();
   }

  return 42;
}

async function bar1(): Promise<number> {
  try {
     return foo(0);
  } catch (err) {
     return -1;
  }
}

async function bar2(): Promise<number> {
  try {
     return await foo(0);
  } catch (err) {
     return -1;
  }
}

```
the function bar1() is probably a bug as it does not call await and the catch block will not be handled in the context of the function as intended while bar2() function is the current implementation.

eslint should probably warn about a potential error when evaluating bar1()

**Comment by bradzacher** [comment]
Check out the `return-await` rule

https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/return-await.md
